server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +696,46 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog (effectively moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo}) for terms below <code>belowTerm</code>
+     * and seq# above <code>aboveSeqNo</code>
+     */
+    public void trim(long belowTerm, long aboveSeqNo) throws IOException {
+        if (aboveSeqNo < 0) {
```

Comment:
Oh, now I see where the protection is. I don't think we  need here so deep. We can remove it here and put it on the resync action as I suggested.